src/show_dialog.cpp

```diff
@@ -201,7 +201,7 @@ dialog_frame::dimension_measurements dialog_frame::layout(int x, int y, int w, i
 	}
 
 	y -= dim_.title.h;
-	w = std::max<int>(w,std::max<int>(int(dim_.title.w),int(buttons_width)));
+	w = std::max<int>(w,std::max<int>(dim_.title.w,int(buttons_width)));
```

Comment:
Huh? The target type is definitely `int`. `w` is `int` (line 179), and so is `dim_.title.w` (`SDL_Rect::w`, see http://wiki.libsdl.org/SDL_Rect ). `Buttons_width` is cast to `int`. Finally, the return value is assigned to `w` that is an `int`.